### ISSUE TITLE
docker-containerd can not build new docker-containerd-shim after killed docker-containerd-shim process

### ISSUE THREAD

**Issue by seven-eleven** [moby/moby#36559]
**Description**

docker-containerd can not build new docker-containerd-shim after killed docker-containerd-shim process.
I killed docker-containerd-shim process. 'docker ps' can list the container, but 'docker exec' failed. Later, docker-containerd didn't build a new docker-containerd-shim.

**Steps to reproduce the issue:**
1. Kill a docker-containerd-shim or all
2. Exec 'docker ps' to get a container
3. Exec 'docker exec -it xxx bash', failed
4. Exec 'ps -ef | grep docker-container-shim', you will see the num less than 'docker ps' for a long time.

**Describe the results you received:**
Exec 'docker exec -it xxx bash', failed
Exec 'ps -ef | grep docker-container-shim', you will see the num less than 'docker ps' for a long time.

**Describe the results you expected:**
Exec 'docker exec -it xxx bash', success
Exec 'ps -ef | grep docker-container-shim', the shim number is the same as 'docker ps' containers number

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:       17.12.1-ce
 API version:   1.35
 Go version:    go1.9.4
 Git commit:    7390fc6
 Built: Tue Feb 27 22:17:57 2018
 OS/Arch:       linux/amd64

Server:
 Engine:
  Version:      17.12.1-ce
  API version:  1.35 (minimum version 1.12)
  Go version:   go1.9.4
  Git commit:   7390fc6
  Built:        Tue Feb 27 22:16:30 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

**Output of `docker info`:**

```
Containers: 164
 Running: 81
 Paused: 0
 Stopped: 83
Images: 9
Server Version: 17.12.1-ce
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9b55aab90508bd389d7654c4baf173a981477d55
runc version: 9f9c96235cc97674e935002fc3d78361b696a69e
init version: 949e6fa
Kernel Version: 4.9.0-0.bpo.5-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 40
Total Memory: 251.7GiB
Name: shd-t-docker-02
ID: 36KR:4EZJ:5VYA:PU4I:2PVK:5FK2:3CGI:2UUU:VTVC:YNZY:C5GC:72J5
Docker Root Dir: /data/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: true
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Linux shd-t-docker-02 4.9.0-0.bpo.5-amd64 #1 SMP Debian 4.9.65-3+deb9u2~bpo8+1 (2017-01-05) x86_64 GNU/Linux
Dockerd args: /usr/bin/dockerd -H fd:// -s overlay2 --data-root=/data/docker --bip=10.10.2.10/24 --log-opt max-size=50m --live-restore

**Comment by seven-eleven** [comment]
@thaJeztah I think that docker-containderd will rebuild docker-containerd-shim, is it right ?

**Comment by cpuguy83** [comment]
No, containerd does not re-create the shim.
Why did you kill the shim?

**Comment by seven-eleven** [comment]
@cpuguy83 Just for test.
I did the same operations on docker 17.09.1-ce with containerd 0.2.3. When I killed containerd, containerd-shim & container were rebuilt immediately.

**Comment by cpuguy83** [comment]
@seven-eleven Are you sure there isn't a restart-policy associated with the container?
In docker 17.11+ the shim is very different.

**Comment by seven-eleven** [comment]
@cpuguy83 
Case 0: Docker version 17.05.0-ce, kubelet running over, I killed containerd-shim, Then containerd-shim & container was rebuilt. I can see a new containerd-shim and container(by `docker ps` command)
Case 1: I stopped kubelet, killed containerd-shim, containerd-shim & container was not rebuilt with docker version 17.05
I think it was k8s rebuilt the container. But the difference is: 
When containerd-shim was killed, 17.05 or 17.09 version can tell the status of container is not normal, so k8s knows to build a new one; but 17.12 vesion tells the status of container is normal, so k8s did nothing.

**Comment by cpuguy83** [comment]
I cannot reproduce this on master.
As soon as the shim is killed it triggers a cleanup event.

**Comment by cpuguy83** [comment]
Likely not fixed in 17.12.1 because the patch to bring in containerd 1.0.2 wasn't merged in time: https://github.com/docker/docker-ce/pull/434

**Comment by thaJeztah** [comment]
Let me close this ticket for now, as it looks like it went stale.
